tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,322 @@
+image(log(b58), col=grey(0:100/100))
+
+```
+
+Perfect - now we have a spatial subset that includes all of the different land cover types that we are interested in investigating. 
+
+<div id="ds-challenge" markdown="1">
+### Challenge: Pick your subset
+
+1. Pick your own area of interest for this spatial subset, and find the rows and columns that capture that area. Can you include some solar panels, as well as the water body?
+
+2. Does it make a difference if you use a band from another part of the electromagnetic spectrum, such as the near-infrared? Hint: you can use the 'wavelengths' function above while removing the `index=` argument to get the full list of band wavelengths.
+
+</div>
+
+## Extracting the example sbset
+
+Now that we have determined our ideal spectral and spatial subsets for our analysis, we are ready to put both of those pieces of information into our `h5read()` function to extract our example subset out of the full NEON hyperspectral dataset. Here, we are taking every fourth band (using our `idx` variabe), columns 501:1000 (the right half of the tile) and rows 1:500 (the bottom half of the tile). The results in us extracting every fourth band of the bottom-right quadrant of the mosaic tile.
+
+```{r read-hyp-data}
+
+# Read in reflectance data.
+# Note the list that we feed into the index argument! 
+# This tells the h5read() function which bands, rows, and 
+# columns to read. This is ultimately how we reduce the file size.
+hs <- h5read(file = f, 
+             name = "SJER/Reflectance/Reflectance_Data", 
+             index = list(idx, subset_columns, subset_rows)
+            )
+```
+
+As per the 'adding group attributes' section above, we will need to add the attributes to the hyperspectral data (`hs`) before writing to the new HDF5 subset file (`nf`). The `hs` variable already has one attribute, `$dim`, which contains the actual dimensions of the `hs` array, and will be important for writing the array to the `nf` file later. We will want to combine this attribute with all of the other Reflectance_Data group attributes from the original HDF5 file, `f`. However, some of the attributes will no longer be valid, such as the `Dimensions` and `Spatial_Extent_meters` attributes, so we will need to overwrite those before assigning these attributes to the `hs` variable to then write to the `nf` file.
+
+```{r hyp-data-attributes}
+
+# grab the '$dim' attribute - as this will be needed 
+# when writing the file at the bottom
+hsd <- attributes(hs)
+
+# We also need the attributes for the reflectance data.
+ha <- h5readAttributes(file = f, 
+                       name = "SJER/Reflectance/Reflectance_Data")
+
+# However, some of the attributes are not longer valid since 
+# we changed the spatial extend of this dataset. therefore, 
+# we will need to overwrite those with the correct values.
+ha$Dimensions=c(500,500,107) # Note that the HDF5 file saves dimensions in a different order than R reads them
```

Comment:
should the = be changed to <-  assignment operator?

Reply:
Hah yes, this is going to take me a while to get used to using the <- operator..